common/djangoapps/student/models.py

```diff
@@ -1442,6 +1442,28 @@ def enrollments_for_user(cls, user):
         return cls.objects.filter(user=user, is_active=1)
 
     @classmethod
+    def preloaded_enrollments_for_user(cls, user):
+        """
```

Comment:
I'm assuming we will need two calls, or a parameter, so I propose either:
```
def enrollments_for_user(cls, user, preload_existing_course_overviews=False):
or
def enrollments_for_user_with_possible_preload(cls, user):
```
I like @pwnage101's parameter suggestion better because you can get more precise with the name, but I would add `_existing_` (or something) to highlight that it is partial.  Or, `partial_preload_course_overviews`.